test/e2e/network_stress_test.go

```diff
@@ -34,35 +36,59 @@ func TestE2EStress(t *testing.T) {
 
 	var wg sync.WaitGroup
 
-	limiter := rate.NewLimiter(1000, 50)
+	limiter := rate.NewLimiter(rate.Limit(config.targetTPS), 50)
 
 	var mutex sync.Mutex
 	var errors []error
 	var errorTransactionStatuses []string
 
+	var clients []*orbsClient.OrbsClient
+	for _, apiEndpoint := range config.apiEndpoints {
+		clients = append(clients, orbsClient.NewClient(apiEndpoint, uint32(generalConfig.AppVcid), codec.NETWORK_TYPE_TEST_NET))
+	}
+
+	defaultTarget, _ := orbsClient.CreateAccount()
+
 	for i := int64(0); i < config.numberOfTransactions; i++ {
 		if err := limiter.Wait(context.Background()); err == nil {
 			wg.Add(1)
 
-			go func() {
+			go func(i int64) {
 				defer wg.Done()
 
-				target, _ := orbsClient.CreateAccount()
+				target := defaultTarget
+				if !config.skipState {
+					target, _ = orbsClient.CreateAccount()
+				}
+
 				amount := uint64(ctrlRand.Intn(10))
 
-				response, _, err2 := h.SendTransaction(OwnerOfAllSupply.PublicKey(), OwnerOfAllSupply.PrivateKey(), "BenchmarkToken", "transfer", uint64(amount), target.AddressAsBytes())
+				client := clients[i%int64(len(clients))] // select one of the clients
+
+				var response *codec.TransactionResponse
+				var err2 error
+
+				if config.async {
```

Comment:
The use case is that async mode can help you get around the limitations on number of connections because the connections will close faster.

Goroutines serve a different purpose, which is sending concurrent requests.